### PR TITLE
fix: corrected Index cond. for conditional formatting in table widget

### LINKED ISSUE #8839: [Bug]: Sorting the Table Widget breaks conditional formatting
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Re-Sorting the Table Widget causes the conditional formatting (e.g. text color or font style based on the field content) to stop working for certain rows

[Example video](https://youtu.be/Egm102BnAfM)
[Example app](https://app.appsmith.com/applications/617830c3b2db5a27f21af2e4/pages/617830c3b2db5a27f21af2e6)

### Steps To Reproduce

1. Create a table widget and fill it with data
2. Format the text color or font style of a row based on its fields content (e.g. {{currentRow.color == "blue" ? "blue" : black"}})
3. Re-Sort the table

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by keyurparalkar**
## Description

This PR fixes the issue related to the conditional formatting breaking when the sorting is applied. Following changes were made: 

- Changed the condition for `originalIndex` variable so as to include the 0th index from `__originalIndex__` property.

Please refer to the issue below for more details:

Fixes #8839 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

I have written integration test for this issue. Files added:

- `Table_Widget__ConditionalFormatting_spec.js`
- `tableWidgetCondnFormatDsl.json`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/fix-condtitional_formatting_break_table 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.14 **(0)** | 37.08 **(0.01)** | 33.96 **(0)** | 55.65 **(0.01)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :green_circle: | app/client/src/widgets/TableWidget/widget/index.tsx | 29.64 **(0)** | 9.74 **(0.05)** | 29.41 **(0)** | 29.29 **(0)**</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/widget/index.tsx [hunk 1]
```diff
@@ -146,7 +146,9 @@ class TableWidget extends BaseWidget<TableWidgetProps, WidgetState> {
         Cell: (props: any) => {
           const rowIndex: number = props.cell.row.index;
           const data = this.props.filteredTableData[rowIndex];
-          const originalIndex = data?.__originalIndex__ || rowIndex;
+          const originalIndex = !isNil(data?.__originalIndex__)
+            ? data.__originalIndex__
+            : rowIndex;
```
Comment (Tooluloope) [inline app/client/src/widgets/TableWidget/widget/index.tsx:151]: This should work as well

```suggestion
          const originalIndex = data?.__originalIndex__ ?? rowIndex;
```

**Review by Tooluloope — APPROVED** [review]

File: app/client/src/widgets/TableWidget/widget/index.tsx (hunk 1 above)
Comment (keyurparalkar) [reply app/client/src/widgets/TableWidget/widget/index.tsx]: I made the above suggested changes.

**Review by Tooluloope — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@keyurparalkar verified this PR and working as expected now.. The conditional formatting remains on sorting

**Pull request merged**